kube_operator/api_objects.go

```diff
@@ -456,6 +441,27 @@ func (cr CustomResourceV1Beta1) Uninstall(c KubeClient, namespace string) {
 	err = crClient.Namespace(namespace).Delete(newCrName, &metav1.DeleteOptions{})
 	if err != nil {
 		glog.Warningf(kwlog(fmt.Sprintf("unable to delete operator custom resource %s. Error: %v", newCrName, err)))
+	} else {
+		status, err := cr.Status(c, namespace)
+		timeout := 20
+		for (err == nil || status != nil) && timeout > 0 {
+			glog.Infof(kwlog(fmt.Sprintf("Custom Resource %s is not yet down. Pausing for 10 before checking again. Custom resource status is: %v Error is %v", newCrName, status, err)))
+			time.Sleep(10 * time.Second)
+			timeout = timeout - 1
+		}
```

Comment:
Put an error message in the log if the timeout occurs, and in the same code below too. You could make a reusable function for the whole status check too.

Reply:
Fixed in both